include/pybind11/common.h

```diff
@@ -28,6 +28,27 @@
 #  endif
 #endif
 
+// Compiler version assertions
+#if defined(__INTEL_COMPILER)
+// ICC v15 or newer required
+static_assert(__INTEL_COMPILER >= 1500, "pybind11 requires Intel C++ compiler v15 or newer");
```

Comment:
The offending compiler may not actually have `static_assert` -- might be better to use `#error`.

Minor: The comments are a bit redundant as they just say the same thing as the error message and/or assertion statement.